services/motion/builtin/move_request.go

```diff
@@ -189,6 +217,75 @@ func (mr *moveRequest) deviatedFromPlan(ctx context.Context, waypoints state.Way
 }
 
+// getTransientDetections returns a list of geometries as observed by the provided vision service and camera.
+// Depending on the caller, the geometries returned are either in their relative position
+// with respect to the base or in their absolute position with respect to the world.
+func (mr *moveRequest) getTransientDetections(
+	ctx context.Context,
+	visSrvc vision.Service,
+	camName resource.Name,
+	f func(g spatialmath.Geometry) spatialmath.Geometry,
+) (*referenceframe.GeometriesInFrame, error) {
+	mr.logger.CDebugf(ctx,
+		"proceeding to get detections from vision service: %s with camera: %s",
+		visSrvc.Name().ShortName(),
+		camName.ShortName(),
+	)
+	// any obstacles specified by the worldstate of the moveRequest will also re-detected here.
+	// get detections from vision service
+	detections, err := visSrvc.GetObjectPointClouds(ctx, camName.Name, nil)
+	if err != nil {
+		return nil, err
+	}
+	mr.logger.CDebugf(ctx, "got %d detections", len(detections))
+
+	// determine transform of camera to base
+	cameraOrigin := referenceframe.NewPoseInFrame(camName.ShortName(), spatialmath.NewZeroPose())
+	cameraToBase, err := mr.fsService.TransformPose(ctx, cameraOrigin, mr.kinematicBase.Name().ShortName(), nil)
+	if err != nil {
+		// here we make the assumption the camera is coincident with the world
+		mr.logger.CDebugf(ctx,
+			"we assume the base named: %s is coincident with the camera named: %s due to err: %v",
+			mr.kinematicBase.Name().ShortName(), camName.ShortName(), err.Error(),
+		)
+		cameraToBase = cameraOrigin
+	}
+	mr.logger.CDebugf(ctx, "cameraToBase transform: %v", spatialmath.PoseToProtobuf(cameraToBase.Pose()))
+
+	// transformed detections
+	transformedGeoms := []spatialmath.Geometry{}
+	for i, detection := range detections {
+		geometry := detection.Geometry
+		// update the label of the geometry so we know it is transient
+		label := camName.ShortName() + "_transientObstacle_" + strconv.Itoa(i)
+		if geometry.Label() != "" {
+			label += "_" + geometry.Label()
+		}
+		geometry.SetLabel(label)
+		mr.logger.CDebugf(ctx, "detection %d observed from the camera frame coordinate system: %s - %s",
+			i, camName.ShortName(), geometry.String(),
+		)
+
+		// transform the transform the geometry to be relative to the base frame which is +Y forwards
+		relativeGeom := geometry.Transform(cameraToBase.Pose())
+		mr.logger.CDebugf(ctx, "detection %d observed from the camera's position in the base frame coordinate system has pose: %v",
+			i, spatialmath.PoseToProtobuf(relativeGeom.Pose()),
+		)
+
+		// apply any other transformation on the geometry defined a priori by the caller
+		transformedGeom := f(relativeGeom)
+		mr.logger.CDebugf(ctx, "transformed detection %d has pose: %v",
+			i, spatialmath.PoseToProtobuf(transformedGeom.Pose()),
+		)
+		transformedGeoms = append(transformedGeoms, transformedGeom)
+	}
+
+	return referenceframe.NewGeometriesInFrame(referenceframe.World, transformedGeoms), nil
```

Comment:
i do not think they should be in the frame of the base since `CheckPlan` is now relative. 

My reasoning is that we are treating the obstacles the same way here, as we do at plan construction. 
For plan construction we do:
```go
	geoms := make([]spatialmath.Geometry, 0, len(worldObstacles))
	for _, geom := range worldObstacles {
		geoms = append(geoms, geom.Transform(startPoseInv))
	}

	gif := referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)
	worldState, err := referenceframe.NewWorldState([]*referenceframe.GeometriesInFrame{gif}, nil)
	if err != nil {
		return nil, err
	}
```
And we end up doing

I believe we are doing the same thing thing here.